### ISSUE TITLE
WARNING: Forward slash in path on windows

### ISSUE THREAD

**Issue by oroppas** [graalvm/mx#204]
```
WARNING: Forward slash in path on windows: C:\msys64\home\kazuo\packages\graal\graal\sdk\mxbuild\windows-amd64\GRAALVM_UNKNOWN_JAVA11_STAGE1\graalvm-unknown-java11-20.0.0-dev/languages/nfi/truffle-nfi.src.zip
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 19293, in <module>
    main()
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 19274, in main
    retcode = c(command_args)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx_commands.py", line 147, in __call__
    return self.command_function(*args, **kwargs)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 13566, in build
    t.execute()
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 4660, in execute
    _built = self.build()
  File "C:\msys64\home\kazuo\packages\graal\graal\sdk\mx.sdk\mx_sdk_vm_impl.py", line 795, in build
    super(GraalVmLayoutDistributionTask, self).build()
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 4978, in build
    self.subject.make_archive()
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 5767, in make_archive
    self._install_source(source, output, destination, arc)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 5563, in _install_source
    ), include=source['path'], excludes=source.get('exclude'), optional=source['optional'])
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 5536, in _install_source_files
    merge_recursive(_source_file, _dst, _arcname, excludes, archive=archive)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 5491, in merge_recursive
    absolute_destination = _safe_path(join(output, dst))
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 325, in _safe_path
    traceback.print_stack()
Finished Archiving GRAALVM_UNKNOWN_JAVA11_STAGE1
[skipping LIBFFI_DIST]
WARNING: Forward slash in path on windows: C:\msys64\home\kazuo\packages\graal\graal\truffle\mx.truffle/macro-truffle.properties
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 19293, in <module>
    main()
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 19274, in main
    retcode = c(command_args)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx_commands.py", line 147, in __call__
    return self.command_function(*args, **kwargs)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 13566, in build
    t.execute()
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 4655, in execute
    buildNeeded, reason = self.needsBuild(newestInput)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 5308, in needsBuild
    sup = super(LayoutArchiveTask, self).needsBuild(newestInput)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 4972, in needsBuild
    reason = self.subject.needsUpdate(newestInput)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 5779, in needsUpdate
    up = _needsUpdate(source_file, self.path)
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 274, in _needsUpdate
    if ts.isOlderThan(newestInput):
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 14438, in isOlderThan
    if getmtime(f) > self.timestamp:
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 3819, in getmtime
    return os.path.getmtime(_safe_path(name))
  File "C:\msys64\home\kazuo\packages\graal\mx\mx.py", line 325, in _safe_path
    traceback.print_stack()
```

**Comment by dougxc** [comment]
@ansalond I looked into this briefly and thought I had a simple fix:
```
diff --git a/mx.py b/mx.py
index 8214e6a..00adff8 100755
--- a/mx.py
+++ b/mx.py
@@ -5448,6 +5448,7 @@ class LayoutDistribution(AbstractDistribution):
                 source_dict = LayoutDistribution._as_source_dict(source, distribution_name, destination, path_substitutions, string_substitutions, distribution_object, context)
                 if substs:
                     destination = substs.substitute(destination)
+                destination = destination.replace('/', os.sep)
                 yield destination, source_dict

     def _walk_layout(self):
```
This makes it work for the [paths](https://github.com/oracle/graal/blob/1c8c97cbd9216dce4c69c0cff5d5940f86a76309/truffle/mx.truffle/suite.py#L862-L863) in `TRUFFLE_NFI_NATIVE`. However, it breaks the [paths](LLVM_TOOLCHAIN) in `LLVM_TOOLCHAIN`:
```
Archiving LLVM_TOOLCHAIN... [C:\Users\dsimon\graal\graal\sdk\mxbuild\windows-amd64\dists\llvm-toolchain.tar does not exist]
  File "C:\Users\dsimon\mx\mx.py", line 19341, in <module>
    main()
  File "C:\Users\dsimon\mx\mx.py", line 19322, in main
    retcode = c(command_args)
  File "C:\Users\dsimon\mx\mx_commands.py", line 147, in __call__
    return self.command_function(*args, **kwargs)
  File "C:\Users\dsimon\mx\mx.py", line 13614, in build
    t.execute()
  File "C:\Users\dsimon\mx\mx.py", line 4671, in execute
    _built = self.build()
  File "C:\Users\dsimon\mx\mx.py", line 4989, in build
    self.subject.make_archive()
  File "C:\Users\dsimon\mx\mx.py", line 5779, in make_archive
    self._install_source(source, output, destination, arc)
  File "C:\Users\dsimon\mx\mx.py", line 5596, in _install_source
    context=self)
  File "C:\Users\dsimon\mx\mx.py", line 3758, in abort
    traceback.print_stack()
  File "C:\Users\dsimon\graal\graal\sdk\mx.sdk\suite.py", line 343 in definition of LLVM_TOOLCHAIN:
Invalid source 'extracted-dependency:LLVM_ORG_COMPILER_RT_LINUX' used in destination '.\':
When using 'extracted-dependency' to extract to a single file, a path must be specified. Did you mean
 - '.\/' as a destination (i.e., extracting all files from 'LLVM_ORG_COMPILER_RT_LINUX' into .\)
 - or 'extracted-dependency:LLVM_ORG_COMPILER_RT_LINUX/path/to/file/in/archive' as a source (i.e., extracting /path/to/file/in/archive from 'LLVM_ORG_COMPILER_RT_LINUX' to '.\')
```

**Comment by ansalond** [comment]
Some time ago we merged some changes to mx (graalvm/mx@94903a9) and graal (580ba3f). The issue should now be fixed.
